content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md

```diff
@@ -431,3 +431,102 @@ See [Enabling signed kubelet serving certificates](/docs/tasks/administer-cluste
+[upgrade/etcd] Waiting for previous etcd to become available
+I0907 10:10:09.109104    3704 etcd.go:588] [etcd] attempting to see if all cluster endpoints ([https://172.17.0.6:2379/ https://172.17.0.4:2379/ https://172.17.0.3:2379/]) are available 1/10
+[upgrade/etcd] Etcd was rolled back and is now available
+static Pod hash for component etcd on Node kinder-upgrade-control-plane-1 did not change after 5m0s: timed out waiting for the condition
+couldn't upgrade control plane. kubeadm has tried to recover everything into the earlier state. Errors faced
+k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade.rollbackOldManifests
+	cmd/kubeadm/app/phases/upgrade/staticpods.go:525
+k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade.upgradeComponent
+	cmd/kubeadm/app/phases/upgrade/staticpods.go:254
+k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade.performEtcdStaticPodUpgrade
+	cmd/kubeadm/app/phases/upgrade/staticpods.go:338
+k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade.StaticPodControlPlane
+	cmd/kubeadm/app/phases/upgrade/staticpods.go:465
+k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade.PerformStaticPodUpgrade
+	cmd/kubeadm/app/phases/upgrade/staticpods.go:617
+k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.PerformControlPlaneUpgrade
+	cmd/kubeadm/app/cmd/upgrade/apply.go:216
+k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.runApply
+	cmd/kubeadm/app/cmd/upgrade/apply.go:156
+k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.newCmdApply.func1
+	cmd/kubeadm/app/cmd/upgrade/apply.go:74
+github.com/spf13/cobra.(*Command).execute
+	vendor/github.com/spf13/cobra/command.go:940
+github.com/spf13/cobra.(*Command).ExecuteC
+	vendor/github.com/spf13/cobra/command.go:1068
+github.com/spf13/cobra.(*Command).Execute
+	vendor/github.com/spf13/cobra/command.go:992
+k8s.io/kubernetes/cmd/kubeadm/app.Run
+	cmd/kubeadm/app/kubeadm.go:50
+main.main
+	cmd/kubeadm/kubeadm.go:25
+...
+````
+
+The reason for this is that the [patch 118867](https://github.com/kubernetes/kubernetes/pull/118867) accidently imported the package of `k8s.io/kubernetes/pkg/apis/core/v1`, and leads to the generation of default attributes of etcd pod for the kubeadm patch version where the change exists,
+this will result to a diff for the manifest comparison, and kubeadm will expect a change of pod hash while kubelet will never update the hash.
+
+The effected kubeadm version is v1.28.0, v1.28.1 and v1.28.2. Kubeadm has patched the v1.28.3, see [patch 120605](https://github.com/kubernetes/kubernetes/pull/120605), so the same issue shouldn't be seen for the upgrade between the version that is after v1.28.3.
+
+There are two way to workaround this issue if you see the same problem in your cluster.
+- skip the etcd upgrade
+If you hit the same issue, etcd is not expected to be upgraded as there is no change except the default attribute which should be ignored instead, so you can skip it as this, 
+```bash
+kubeadm upgrade apply [version] --etcd-upgrade=false
+```
+
```

Comment:
```suggestion
This is not recommended in case a new etcd version was introduced by a later v1.28 patch version.

```

Reply:
I think if there is a new etcd version, the user will not hit this issue and thus won't follow this trouble shooting item, anyway, updated.